godot-macros/src/lib.rs

```diff
@@ -454,6 +456,60 @@ pub fn derive_from_variant(input: TokenStream) -> TokenStream {
     translate(input, derive_from_variant::transform)
 }
 
+/// Derive macro for [Property](godot::bind::property::Property) on enums.
+///
+/// Currently has some tight requirements which are expected to be softened as implementation expands:
+/// - Only works for enums, structs aren't supported by this derive macro at the moment.
+/// - The enum must have an explicit `#[repr(u*/i*)]` type.
+///     - This will likely stay this way, since `isize`, the default repr type, is not a concept in Godot.
+/// - The enum variants must not have any fields - currently only unit variants are supported.
+/// - The enum variants must have explicit discriminants, that is, e.g. `A = 2`, not just `A`
+///
+/// # Example
+///
+/// ```no_run
+/// # use godot::prelude::*;
+/// #[repr(i32)]
+/// #[derive(Property)]
+/// # #[derive(PartialEq, Eq, Debug)]
+/// enum TestEnum {
+///     A = 0,
+///     B = 1,
+/// }
+///
+/// #[derive(GodotClass)]
+/// struct TestClass {
+///     #[var]
+///     foo: TestEnum
+/// }
+///
+/// # #[godot_api]
+/// # impl TestClass {}
```

Comment:
Could you add a comment:
```suggestion
/// # // TODO remove when https://github.com/godot-rust/gdext/issues/187 is truly addressed
/// # #[godot_api]
/// # impl TestClass {}
```

Otherwise we keep tech-debt around forever  🙂